### PR TITLE
Improve QBean alias handling

### LINKED ISSUE #1379: Projections.bean discards aliases required for sorting
```java
final List<Expression<?>> list = new ArrayList<>();
final NumberPath<Double> avgAmountPath = Expressions.numberPath(Double.class, "avgAmount");
list.add(QEntity.entity.amount.avg().as(avgAmountPath));
jpaQuery.groupBy(QEntity.entity.something);
jpaQuery.orderBy(avgAmountPath.asc());

// this throws exception: Caused by: com.mysql.jdbc.exceptions.jdbc4.MySQLSyntaxErrorException: Unknown column 'avgAmount' in 'order clause'
jpaQuery.list(Projections.bean(MyResult.class, list.toArray(new Expression[0])));

// this is ok
jpaQuery.list(list.toArray(new Expression[0]));
```

### PR DESCRIPTION
**Pull request by timowest**
Fixes #1379

### CODE REVIEW

File: querydsl-core/src/main/java/com/querydsl/core/types/QBean.java [hunk 1]
```diff
@@ -73,6 +78,10 @@
         return rv.build();
     }
 
+    private static boolean isAliasSplitted(Expression<?> expr) {
```
Comment (johnktims) [inline querydsl-core/src/main/java/com/querydsl/core/types/QBean.java]: I don't understand why an expression that is an instance of `FactoryExpression` or  `GroupExpression` is considered a split alias. I think the notion of splitting is confusing me.

Comment (timowest) [inline querydsl-core/src/main/java/com/querydsl/core/types/QBean.java]: FactoryExpressions and GroupExpressions are compound expressions which are serialized as individual items, so you can't really alias them on the SQL/JPQL level.

The change introduced in this PR is to serialize the alias into the query, if the left hand side of the alias operation is not a FactoryExpression or GroupExpression.

Comment (timowest) [inline querydsl-core/src/main/java/com/querydsl/core/types/QBean.java]: Shall I rename the method?

**Comment by timowest** [comment]
@johnktims @Shredder121 Is this ok to merge?

File: querydsl-core/src/main/java/com/querydsl/core/types/QBean.java (hunk 1 above)
Comment (johnktims) [inline querydsl-core/src/main/java/com/querydsl/core/types/QBean.java]: Sorry. I forgot to respond to your question. I would rename the method. `isCompoundExpression`?

Comment (timowest) [inline querydsl-core/src/main/java/com/querydsl/core/types/QBean.java]: Sounds good, I'll do that.

Comment (timowest) [inline querydsl-core/src/main/java/com/querydsl/core/types/QBean.java]: Changed the name.

**Comment by johnktims** [comment]
This PR looks good to me. @Shredder121 ?

**Comment by Shredder121** [comment]
Yes, it looks good now.
For the sake of completeness I'd say let the build finish, and then I'll merge it.
Or you can do it as well.

**Pull request merged**